mock/api.js

```diff
@@ -297,7 +282,11 @@ export const mockSession = {
 export default {
   'GET /controllers/months': mockIndices,
   'POST /controllers/list': mockControllers,
-  'POST /datasets/list': mockResults,
+  'POST /datasets/list': (req, res) => {
+    const data = {};
+    data[req.body.controller] = mockResults;
+    res.send(data);
+  },
```

Comment:
This is interesting, because you're taking the `req.body.controller`, the controller name we're requesting (as returned by the `/controllers/list` mock), but returning the list of datasets with randomly named controllers. Really, each element in the mock array should be that same `req.body.controller`. I wonder if it would be better to generate the mock here rather than trying to re-use a static mock value? You're generating the outer JSON dynamically anyway.

Also, really all of the mocks could be done this way so the response matches the request, but that wasn't done with the others. We ought to be consistent. I like the idea of the response matching; it's just odd to do it only here.

Reply:
Fixed the controllers to match. Had to update other queries too.